### DISCUSSION TITLE
Proper Cleanup

### DISCUSSION THREAD

**Discussion by hxorasani** [libuv/libuv#3954] [Q&A] [answered]
I'm using libuv with duktape and my app needs to be able to reload itself live.
is there an official proper way to "cleanup" or "restart" libuv?

**Comment by bnoordhuis** [comment, marked as answer]
Close all handles, cancel or wait for all requests to finish, then delete the event loop.

**Reply by hxorasani** [reply]
What if it's the "uv_default_loop()"? Does that also need to be deleted? I'm confused.

**Reply by bnoordhuis** [reply]
Yes.
